src/reconcile/rec_write.c

```diff
@@ -1843,8 +1843,10 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
             return (__wt_set_return(session, EBUSY));
 
         /* If we need to restore the page to memory, copy the disk image. */
-        if (r->cache_write_restore)
+        if (r->cache_write_restore) {
+            multi->restore = true;
```

Comment:
If we copy the disk image here can't we instead check that we don't have a disk image in rec_hs_wrapup and remove the need for this flag?

Reply:
In rec_hs_wrapup we have already finished writing the disk image and potentially copied the image if we want to restore it. We have to do it here to decide whether to do a restore on this page and copy the image.

Reply:
We have cases that we have disk images but don't want to restore, like scrub eviction.